src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        threshold = calculateThresholdValue();
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(index, key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else if (Objects.equals(table[index].key, newNode.key)
+                && table[index].next == null) {
+            table[index] = newNode;
+        } else if (Objects.equals(table[index].key, newNode.key)
+                && table[index].next != null) {
+            newNode.next = table[index].next;
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (current.next == null) {
+                    current.next = newNode;
+                    size++;
+                    break;
+                }
+                if (Objects.equals(current.next.key, newNode.key)) {
+                    newNode.next = current.next.next;
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexOfKey = getIndex(key);
+        if (isIndexInTheTable(indexOfKey)) {
+            Node<K, V> current = table[indexOfKey];
+            while (current != null) {
+                if (Objects.equals(current.key,key)) {
+                    return current.value;
+                }
+                current = current.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int calculateThresholdValue() {
+        return (int) (currentCapacity * DEFAULT_LOAD_FACTOR);
+    }
+
+    private void resize() {
+        if (size == threshold) {
+            currentCapacity *= GROW_COEFFICIENT;
+            threshold = calculateThresholdValue();
+            size = 0;
+            Node<K, V>[] oldTable = table;
+            table = (Node<K, V>[]) new Node[currentCapacity];
+            for (int i = 0; i < oldTable.length; i++) {
+                if (oldTable[i] == null) {
+                    continue;
+                }
+                Node<K, V> current = oldTable[i];
+                while (current != null) {
+                    int newIndex = getIndex(current.key);
+                    put(current.key, current.value);
+                    current = current.next;
+                }
+            }
+        }
+    }
+
+    private int getIndex(K key) {
+        int index = key == null ? 0 : key.hashCode() % currentCapacity;
+        if (index < 0) {
+            index *= -1;
+        }
+        return index;
```

Comment:
```suggestion
        return key == null ? 0 : key.hashCode() % currentCapacity;;
```

Reply:
Sometimes  (key.hashCode() ) returns negative numbers and we have ArrayIndexOutOfBoundsException. In this case I have to multiply the value by -1.

Reply:
okay use Math.abs(key.hashCode() % currentCapacity) then